pkg/shared/service.go

```diff
@@ -18,6 +22,103 @@ func NewServiceHandler() *ServiceHandler {
 
 func (sh *ServiceHandler) Handle(ctx context.Context, event sdk.Event) error {
 	fmt.Println("handling object ", event.Object.GetObjectKind().GroupVersionKind().String())
+
+	sharedService := event.Object.(*v1alpha1.SharedService)
+	sharedServiceCopy := sharedService.DeepCopy()
+
+	if sharedService.Spec.ServiceType != strings.ToLower(v1alpha1.KeycloakKind) {
+		return nil
+	}
+
+	if event.Deleted {
+		return nil
+	}
+
+	if sharedService.GetDeletionTimestamp() != nil {
+		return sh.finalizeSharedService(sharedServiceCopy)
+	}
+
+	switch sharedService.Status.Phase {
+	case v1alpha1.SSPhaseNone:
+		sh.initSharedService(sharedServiceCopy)
```

Comment:
this switch > case > method is quite tidy. 
@dimitraz @JameelB do you think we should apply this to the keycloak-operator work you've been doing on the deploy?

Reply:
yeah, doing that atm. Agreed it looks a lot better